packages/rum/src/userActionCollection.ts

```diff
@@ -65,57 +32,71 @@ export interface AutoUserAction {
 
 export type UserAction = CustomUserAction | AutoUserAction
 
+export let currentUserActionProcess = { stop: noop }
+
 export function startUserActionCollection(lifeCycle: LifeCycle) {
+  const subscriptions: Subscription[] = []
+
+  addEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
   function processClick(event: Event) {
     if (!(event.target instanceof Element)) {
       return
     }
-
-    newUserAction(lifeCycle, UserActionType.CLICK, getElementContent(event.target))
+    currentUserActionProcess = newUserAction(lifeCycle, UserActionType.CLICK, getElementContent(event.target))
   }
 
-  addEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
-
   return {
     stop() {
+      currentUserActionProcess.stop()
       removeEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
+      subscriptions.forEach((s) => s.unsubscribe())
     },
   }
 }
 
-let currentUserAction: { id: string; startTime: number } | undefined
+interface PendingAutoUserAction {
+  id: string
+  startTime: number
+}
+let currentUserAction: PendingAutoUserAction | undefined
 
-function newUserAction(lifeCycle: LifeCycle, type: UserActionType, name: string) {
+function newUserAction(lifeCycle: LifeCycle, type: UserActionType, name: string): { stop(): void } {
   if (currentUserAction) {
-    // Discard any new user action if another one is already occuring.
-    return
+    // Discard any new click user action if another one is already occuring.
+    return { stop: currentUserActionProcess.stop }
   }
 
   const id = generateUUID()
   const startTime = performance.now()
   currentUserAction = { id, startTime }
 
-  const { observable: pageActivitiesObservable, stop: stopPageActivitiesTracking } = trackPageActivities(lifeCycle)
   const { eventCounts, stop: stopEventCountsTracking } = trackEventCounts(lifeCycle)
 
-  waitUserActionCompletion(pageActivitiesObservable, (endTime) => {
-    stopPageActivitiesTracking()
+  waitIdlePageActivity(lifeCycle, (endTime) => {
     stopEventCountsTracking()
-    if (endTime !== undefined) {
+    if (endTime !== undefined && currentUserAction !== undefined && currentUserAction.id === id) {
       lifeCycle.notify(LifeCycleEventType.USER_ACTION_COLLECTED, {
         id,
         name,
-        startTime,
         type,
-        duration: endTime - startTime,
+        duration: endTime - currentUserAction.startTime,
```

Comment:
Same comment as https://github.com/DataDog/browser-sdk/pull/388#discussion_r419483686 , I don't see why we use `currentDuration.startTime` instead of `startTime` directly.